lib/Interpreter/CppInterOp.cpp

```diff
@@ -2768,18 +2939,36 @@ namespace Cpp {
     return getInterp().toString(type, obj);
   }
 
-  static QualType InstantiateTemplate(TemplateDecl* ClassDecl,
-                                      TemplateArgumentListInfo& TLI, Sema &S) {
-    // This will instantiate tape<T> type and return it.
-    SourceLocation noLoc;
-    QualType TT = S.CheckTemplateIdType(TemplateName(ClassDecl), noLoc, TLI);
-
+  static Decl* InstantiateTemplate(TemplateDecl* TemplateD,
+                                   TemplateArgumentListInfo& TLI, Sema& S) {
     // This is not right but we don't have a lot of options to choose from as a
     // template instantiation requires a valid source location.
     SourceLocation fakeLoc = GetValidSLoc(S);
+    if (auto* FunctionTemplate = dyn_cast<FunctionTemplateDecl>(TemplateD)) {
+      FunctionDecl* Specialization = nullptr;
+      clang::sema::TemplateDeductionInfo Info(fakeLoc);
+      if (Sema::TemplateDeductionResult Result = S.DeduceTemplateArguments(
+              FunctionTemplate, &TLI, Specialization, Info,
+              /*IsAddressOfFunction*/ true)) {
+        // FIXME: Diagnose what happened.
+        (void)Result;
+      }
+      return Specialization;
```

Comment:
Where this code comes from? Some other PR?

Reply:
Yes, I think I pulled this at some point a month ago but not sure why it doesn't show up as a commit